homeassistant/components/hausbus/config_flow.py

```diff
@@ -0,0 +1,42 @@
+"""Config flow for Haus-Bus integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_SCHEMA = vol.Schema({})
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for hausbus."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            return self.async_create_entry(title="Haus-Bus", data=user_input)
```

Comment:
We should try to discover haus-bus devices and only create the config entry if devices were found.
You can use `self.async_show_progress` to inform the user we're waiting for devices to be discovered.

Reply:
I did add a simple discovery in the config_flow. Now it will only add the hausbus config entry, if at least one device has been discovered.